### PR TITLE
DirectTypescriptBuilder implemented and minor changes made

### PR DESCRIPTION
**Pull request by DovydasNavickas**
Direct Typescript builder implemented to have a faster build pipeline.
Cold full build old times:
```
Typescript: 22s
TSLint: 7s - 9s
```
Hot full build old times:
```
Typescript: 11s - 14s (50%-63% compared to old full cold)
TSLint: 5s -7s (71%-77% compared to old full cold)
```

Cold full build updated times:
```
Typescript: 7.4s
TSLint: 4.4s
```

Hot full build updated times:
```
Typescript: 6s - 6.6s (81%-89% compared to updated full cold)
TSLint: 3.9s - 4.4s (88.6%-100% compared to updated full cold)
```

Also, a `--fast` mode introduced.
Fast times:
```
Typescript: 956ms - 1045ms
Compared to updated cold: 12.91%-14.12%
Compared to updated hot: 15.93%-15.83%

TSLint: 11ms - 14ms
Compared to updated cold: 0.38%
Compared to updated hot: 0.28%-0.38%
```

Fast mode builds only changed files and does no full build therefore corner-cases could occur, e.g. updating const enum values will not update all their usages and full build will be required.
But for most of the development, fast mode will speed things up to 14 times and will make whole build task to take `~1s` instead of `14s` (TS + TSLint combined).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,14 +26,15 @@
...
-  "license": "GPL-3.0",
+  "license": "AGPL-3.0",
   "files": [
     "dist",
     "*.md"
   ],
   "devDependencies": {
     "@types/gulp-cached": "0.0.29",
     "@types/node": "^7.0.5",
+    "multi-glob": "^1.0.1",
```
Comment (GiedriusGrabauskas) [inline package.json]: Move it to the `dependencies`

File: src/builders/typescript/typescript-direct-builder.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                return file.substr(file.length - dtsLength) !== dts;
+            });
+        }
+
+        return await this.Lint(globbedFiles);
+    }
+
+    public PrintDiagnostics(diagnostics: ts.Diagnostic[], logger: Logger) {
+        for (let diagnostic of diagnostics) {
+            if (diagnostic.file == null || diagnostic.file.isDeclarationFile) {
```
Comment (GiedriusGrabauskas) [inline src/builders/typescript/typescript-direct-builder.ts]: Missing check from `tsconfig.json` (`skipLibCheck`).

File: src/builders/typescript/typescript-direct-builder.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import * as path from "path";
+import * as ts from "typescript";
+import * as Linter from "tslint";
+import { glob } from "multi-glob";
+import { ILinterOptionsRaw, LintResult } from "tslint/lib/lint";
+import Configuration from "../../configuration/configuration";
+import { Logger } from "../../utils/logger";
+
+interface TsConfig {
+    compilerOptions: any;
```
Comment (GiedriusGrabauskas) [inline src/builders/typescript/typescript-direct-builder.ts]: Change `any` to `ts.CompilerOptions`

Comment (DovydasNavickas) [inline src/builders/typescript/typescript-direct-builder.ts]: It's not  `ts.CompilerOptions`. That is why [L29](https://github.com/QuatroCode/simplr-gulp/pull/40/files/cee3801b1d9e879b092d792c5886b9a2e6c14abc#diff-9675c861591be1df0ef841ed36d9875cR29) is needed:
```ts
// Parse compiler options (e.g. jsx: react -> jsx: 2)
let parsedCompilerOptions = ts.convertCompilerOptionsFromJson(tsConfigFromJson.compilerOptions, process.cwd());``

File: package.json (hunk 1 above)
Comment (DovydasNavickas) [inline package.json]: https://github.com/QuatroCode/simplr-gulp/pull/40/commits/b9bdc0666d390ec2fee0ef0c62382456256a13d5 Done.

File: src/builders/typescript/typescript-direct-builder.ts (hunk 2 above)
Comment (DovydasNavickas) [inline src/builders/typescript/typescript-direct-builder.ts]: https://github.com/QuatroCode/simplr-gulp/pull/40/commits/502effe881e642a13e1ef3e389e64f360eca79c3 Done.
Now `tsConfig.json` value for `skipDefaultLibCheck` is respected and used while printing diagnostics and definition files are not filtered out while linting, to have all of them if needed.

**Comment by DovydasNavickas** [comment]
Added uglifying for production build. Watch scripts never uglifies.

**Comment by GiedriusGrabauskas** [comment]
Renamed flag from `--no-build` to `--noBuild`.

**Pull request merged**